src/components/internals/rich-text-body/rich-text-body.js

```diff
@@ -252,6 +252,20 @@ const RichTextEditorBody = React.forwardRef((props, ref) => {
     );
   }
 
+  const { focus } = props.editor;
+
+  const onToolbarClick = React.useCallback(() => {
+    // if clicking on a dropdown, we need to both focus the RichTextInput
+    // and open the dropdown.
+    // if we don't use setTimeout, our dropdown opening will be hijacked by the
+    // editors rerendering.
+    // the reason we keep this here, is because any onClick from our dropdown or our mark buttons
+    // will propogate to here.
+    if (!props.editor.value.selection.isFocused) {
+      setTimeout(focus, 0);
```

Comment:
setTimeout allows our dropdown button onClick to "win" the race condition

Reply:
Just curious, would this also work with `setImmediate(focus)` instead of `setTimeout(focus, 0)`? If so, it should be slightly more performant.

Reply:
<img width="677" alt="CleanShot 2019-10-29 at 14 03 11@2x" src="https://user-images.githubusercontent.com/1877073/67769486-dc326d80-fa54-11e9-8055-6e1767eada4f.png">

Reply:
So we want the focus to happen on the next tick essentially?

Reply:
@tdeekens yeah. There are lots of questions like this in the Slate slack channel. Usually people recommend to use setTimeout.

Reply:
https://caniuse.com/#feat=setimmediate

Reply:
I would not use it.

Reply:
![Screen Shot 2019-10-29 at 2 06 20 PM](https://user-images.githubusercontent.com/2425013/67769754-4f3be400-fa55-11e9-8574-70d4a012b3a1.png)
No chrome support

Reply:
![Screen Shot 2019-10-29 at 2 06 51 PM](https://user-images.githubusercontent.com/2425013/67769801-64b10e00-fa55-11e9-8435-333595214966.png)
No Chrome support...

Reply:
I also wouldn't worry about performance. This only happens when 

A) The editor is out of focus
B) The user clicks on a mark button or a dropdown button.

Reply:
Also like `const = scheduleFocusAfter...` would also be cool.

Reply:
Err, I added like 5 lines of comments? I can add more...

Reply:
Oh wow, I had a completely wrong idea about `setImmediate` on the browser 🤦‍♂ This ain't node 😅